### PR TITLE
feat: add endpoint pre-producer privacy gate

### LINKED ISSUE #267: Endpoint: executable pre-producer privacy and threat-model gate
## Objective

Create the repository-only prerequisite gate required before any opt-in endpoint observation producer may be implemented. The gate must make the current privacy and threat-model boundary machine-readable and fail closed in Security CI without collecting endpoint data or granting runtime authority.

## Acceptance Criteria

- [ ] Add one canonical, versioned, machine-readable endpoint producer privacy/threat-model artifact with exact actors, protected data classes, allowed GH-264 output fields, prohibited raw-data classes, consent/least-privilege requirements, retention/transport limits, and residual promotion gates.
- [ ] Keep endpoint collection, OS sensors, correlation, warning, transport, containment, automatic actions, production authority and producer implementation explicitly false or absent.
- [ ] Add a standard-library verifier that rejects unknown/missing fields, malformed types, weakened invariants, schema/ledger/corpus drift, missing forbidden-field evidence, and unauthorized stage advancement with stable data-minimal errors.
- [ ] Add adversarial mutation tests for every protected invariant and safe controls for truthful negative wording.
- [ ] Wire the verifier and tests into the existing Security Audit workflow without adding dependencies.
- [ ] Synchronize README, Whitepaper, Roadmap, FAQ, Landing/public HTML, llms.txt, Memory and Bridge with an exact repository-only/non-production classification.
- [ ] Run complete relevant local checks, independent read-only review, protected PR checks, exact-main CI/Security/Pages and live public readback.

## Boundaries

This issue adds no endpoint producer, telemetry collection, process/file/network/credential/model access, detection truth, AI/actor attribution, warning, transport, proof, response action, wallet, chain, contract, Guardian service, infrastructure, deployment, Mainnet or production behavior. It does not change KAS/PROM, Guardian reputation, slash access control, Commit-Reveal or the no-emergency-stop rule. The repository gate is a review and drift-control mechanism, not proof that arbitrary code contains no hidden sensor and not runtime authorization.

## Work allocation

Kimi K3 supplies a bounded secret-free read-only architecture/privacy analysis. Codex Sol owns scope, architecture and security decisions, implementation, integration, tests, public claims, protected delivery and closeout.

### PR DESCRIPTION
**Pull request by NeaBouli**
## Summary

- add a versioned machine-readable pre-producer endpoint privacy/threat model
- add a dependency-free fail-closed verifier and 35 adversarial mutation tests
- bind the gate to GH-264 corpus/status limits and must-stay-false capabilities
- enforce the gate in Security Audit and synchronize public Markdown, HTML, module docs, status and Bridge surfaces

## Boundaries

- repository drift control only
- no producer, endpoint collection, OS access, retention, transport, proof, response, automation, wallet, chain, deployment or production authority
- does not prove privacy, anonymity or absence of hidden collection code
- KAS/PROM, Guardian L1 reputation, slash ACL, Commit-Reveal and no-emergency-stop invariants are unchanged

## Verification

- `python3 scripts/verify_endpoint_producer_privacy_gate.py`
- `python3 -m unittest scripts.test_endpoint_producer_privacy_gate` (35 passed)
- `python3 scripts/verify_public_claim_consistency.py`
- `python3 -m unittest scripts.test_public_claim_consistency` (67 passed)
- documentation hygiene (11 passed), project status (7 passed), H-001 evidence (4 passed), Memory integrity, Ruff, compile, JSON/HTML and diff checks passed
- Kimi K3 independent review: no P0/P1/P2; one P3 fixed and focused follow-up PASS

Closes #267

## Summary by CodeRabbit

- **New Features**
  - Added a fail-closed Security CI privacy gate for the planned endpoint-producer boundary.
  - Added a machine-readable threat model covering opt-in, least privilege, data minimization, prohibited data, retention, transport, and independent review requirements.
  - Added validation to detect drift in privacy requirements, public status claims, and approved endpoint-observation fields.

- **Documentation**
  - Updated project, architecture, roadmap, FAQ, and module documentation with the new privacy boundary and safeguards.
  - Clarified that no endpoint collection, runtime behavior, response authority, or production capability is added.

### CODE REVIEW

File: .github/workflows/security-audit.yml [hunk 1]
```diff
@@ -37,6 +37,11 @@ jobs:
           python3 scripts/check_public_documentation_hygiene.py
           python3 -m unittest scripts/test_public_documentation_hygiene.py
 
+      - name: Endpoint Producer Privacy Gate
+        run: |
+          python3 scripts/verify_endpoint_producer_privacy_gate.py
+          python3 -m unittest scripts/test_endpoint_producer_privacy_gate.py
```
[the remark this answers is not in the document]
Comment (NeaBouli) [reply .github/workflows/security-audit.yml]: Fixed in 66452c8: Security Audit now runs both the dedicated GH-267 gate and the complete public-claim verifier/test suite.

File: scripts/test_public_claim_consistency.py [hunk 2]
```diff
@@ -611,6 +611,113 @@ def test_gh_264_public_boundary_marker_is_required(self) -> None:
...
+            "GH-267 is production-ready.",
+        )
+        for claim in claims:
+            with self.subTest(claim=claim):
+                normalized = " ".join(claim.split()).casefold()
+                self.assertTrue(
+                    any(
+                        pattern.search(normalized)
+                        for pattern in MODULE.GH267_PROHIBITED_CLAIMS
+                    )
```
[the remark this answers is not in the document]
Comment (NeaBouli) [reply scripts/test_public_claim_consistency.py]: Fixed in 66452c8: representative prohibited claims are appended to every copied GH-267 public surface and exercised end-to-end through MODULE.verify.

File: scripts/verify_endpoint_producer_privacy_gate.py [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+    "hidden_sensor_absence_proven",
+)
+
+
+def load_json(path: Path) -> tuple[Any, str | None]:
+    """Load one JSON file or return a stable error category."""
+    try:
+        text = path.read_text(encoding="utf-8")
+    except OSError:
+        return None, "missing or unreadable"
```
[the remark this answers is not in the document]
Comment (NeaBouli) [reply scripts/verify_endpoint_producer_privacy_gate.py]: Fixed in 66452c8: non-UTF-8 reads return the stable missing-or-unreadable category; a direct regression covers the case.

File: scripts/verify_endpoint_producer_privacy_gate.py [hunk 4]
```diff
@@ -0,0 +1,549 @@
...
+            errors.append("GH-258 must remain non-collecting and non-actioning")
+    if not isinstance(gh_264, dict):
+        errors.append("GH-264 status record missing or invalid")
+    else:
+        if (
+            gh_264.get("status")
+            != "repository_candidate_implemented_and_locally_tested"
+            or gh_264.get("classification")
+            != "canonical_observe_only_endpoint_statement"
+            or gh_264.get("schema_version") != 1
```
[the remark this answers is not in the document]
Comment (NeaBouli) [reply scripts/verify_endpoint_producer_privacy_gate.py]: Fixed in 66452c8: GH-264 and corpus schema versions now require a real JSON integer, with boolean regressions in both dedicated and public status suites.

File: scripts/verify_public_claim_consistency.py [hunk 5]
```diff
@@ -394,6 +394,48 @@
...
 
+GH267_PUBLIC_FILES = GH264_PUBLIC_FILES
+
+GH267_REQUIRED_FRAGMENTS = (
+    "GH-267",
+    "repository",
+    "privacy",
+    "runtime",
+    "production",
+)
```
[the remark this answers is not in the document]
Comment (NeaBouli) [reply scripts/verify_public_claim_consistency.py]: Fixed in 66452c8: GH-267 boundaries now require semantic negative privacy, runtime, and production wording; an end-to-end generic-fragment bypass regression fails closed.

**Comment by NeaBouli** [comment]
Maintainer checkpoint: head `5960f1b` remains fully green and all prior review threads are resolved. No GH-267 defect was found during this intake.

The PR is temporarily held as Draft because a new post-check advisory, `RUSTSEC-2026-0285`, now fails the repository Dependency Audit on unchanged main (`rustls 0.23.41`; fix `>= 0.23.45`). Resolve #272 first, then update this branch onto exact main and rerun all required checks before normal merge. This is a dependency-evidence hold, not a rollback of the GH-267 implementation.
